standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/columnstats/aggr/LongColumnStatsAggregatorTest.java

```diff
@@ -0,0 +1,289 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.hadoop.hive.metastore.columnstats.aggr;
+
+import org.apache.hadoop.hive.common.ndv.fm.FMSketch;
+import org.apache.hadoop.hive.common.ndv.hll.HyperLogLog;
+import org.apache.hadoop.hive.metastore.StatisticsTestUtils;
+import org.apache.hadoop.hive.metastore.TableType;
+import org.apache.hadoop.hive.metastore.annotation.MetastoreUnitTest;
+import org.apache.hadoop.hive.metastore.api.ColumnStatistics;
+import org.apache.hadoop.hive.metastore.api.ColumnStatisticsData;
+import org.apache.hadoop.hive.metastore.api.ColumnStatisticsObj;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.utils.MetaStoreServerUtils.ColStatsObjWithSourceInfo;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.apache.hadoop.hive.metastore.StatisticsTestUtils.assertLongStats;
+import static org.apache.hadoop.hive.metastore.StatisticsTestUtils.createFMSketch;
+import static org.apache.hadoop.hive.metastore.StatisticsTestUtils.createHll;
+import static org.apache.hadoop.hive.metastore.Warehouse.DEFAULT_CATALOG_NAME;
+
+@Category(MetastoreUnitTest.class)
+public class LongColumnStatsAggregatorTest {
+
+  private static final Table TABLE = new Table("dummy", "db", "hive", 0, 0,
+      0, null, null, Collections.emptyMap(), null, null,
+      TableType.MANAGED_TABLE.toString());
+  private static final FieldSchema COL = new FieldSchema("col", "int", "");
+
+  @Test
+  public void testAggregateSingleStat() throws MetaException {
+    LongColumnStatsAggregator aggregator = new LongColumnStatsAggregator();
+
+    List<String> partitionNames = Collections.singletonList("part1");
+
+    HyperLogLog hll = createHll(1, 3);
+    ColumnStatisticsData data1 = StatisticsTestUtils.createLongStats(
+        1L, 2L, 1L, 4L, hll);
+    ColumnStatistics stats1 = StatisticsTestUtils.createColStats(data1, TABLE, COL, partitionNames.get(0));
+
+    List<ColStatsObjWithSourceInfo> statsList = Collections.singletonList(
+        new ColStatsObjWithSourceInfo(stats1.getStatsObj().get(0), DEFAULT_CATALOG_NAME, TABLE.getDbName(),
+            TABLE.getTableName(), partitionNames.get(0)));
+
+    ColumnStatisticsObj stats = aggregator.aggregate(statsList, partitionNames, true);
+
+    assertLongStats(stats, 1L, 2L, 1L, 4L, hll);
```

Comment:
What do you think of using `assertEquals` with `ColumnStatisticsData` objects instead?

The failure message is usually clear and renders nicely by IDE and maven. 

If there are more than one differences in the expected output this would be visible in the message (we don't stop in the first failure).

Programmers can understand directly what is asserted without the need to jump into the method.

Reply:
As stated in another comment, I have re-worked the tests to use plain `Assert.assertEquals` + the builder.